src/dymaptic.GeoBlazor.Core/Components/Layers/CSVLayer.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.AspNetCore.Components;
+using System.Text.Json.Serialization;
+
+
+namespace dymaptic.GeoBlazor.Core.Components.Layers;
+
+/// <summary>
+///     The CSVLayer is a point layer based on a CSV file (.csv, .txt). CSV is a plain-text file format used to
+///     represent tabular data, including geographic point features (latitude, longitude). Typically the latitude
+///     coordinate is the Y value, and the longitude coordinate is the X value. The X, Y coordinates must be stored
+///     in SpatialReference.WGS84 in csv feed.
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-CSVLayer.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class CSVLayer : Layer
+{
+    /// <summary>
+    ///     Parameterless constructor for use as a razor component
+    /// </summary>
+    public CSVLayer()
+    {
+    }
+
+    /// <summary>
+    ///     Constructor for use in code
+    /// </summary>
+    /// <param name="url">
+    ///     The url for the CSV source data.
+    /// </param>
+    /// <param name="title">
+    ///     The title of the layer used to identify it in places such as the Legend and LayerList widgets.
+    /// </param>
+    /// <param name="opacity">
+    ///     The opacity of the layer.
+    /// </param>
+    /// <param name="visible">
+    ///     Indicates if the layer is visible in the View. When false, the layer may still be added to a Map instance that is
+    ///     referenced in a view, but its features will not be visible in the view.
+    /// </param>
+    /// <param name="listMode">
+    ///     Indicates how the layer should display in the LayerList widget. The possible values are listed below.
+    /// </param>
+    public CSVLayer(string url, string? title = null,
+        double? opacity = null, bool? visible = null, ListMode? listMode = null)
+    {
+        Url = url;
+        Title = title;
```

Comment:
I realize this is missing in a _lot_ of classes, including most layers, but we should wrap these parameter setters in `#pragma warning disable BL0005`. Search for that and you'll see some examples where I've implemented it. If you don't do this, and you create the class in C# with this constructor, you will get a warning about setting parameters outside of markup.